holoviews/ipython/__init__.py

```diff
@@ -146,15 +146,22 @@ class notebook_extension(param.ParameterizedFunction):
                  'plotly': 'plotly'}
 
     def __call__(self, *args, **params):
-        imports = [(name, b) for name, b in self._backends.items()
-                   if name in args or params.get(name, False)]
-        if not imports or 'matplotlib' not in Store.renderers:
-            imports = imports + [('matplotlib', 'mpl')]
+        # Get requested backends
+        imports = [(arg, self._backends[arg]) for arg in args
+                   if arg in self._backends]
+        for p, val in sorted(params.items()):
+            if p in self._backends:
+                imports.append((p, self._backends[p]))
+        if not imports:
+            imports = [('matplotlib', 'mpl')]
 
         args = list(args)
+        selected_backend = None
         for backend, imp in imports:
             try:
                 __import__('holoviews.plotting.%s' % imp)
+                if selected_backend is None:
+                    selected_backend = backend
```

Comment:
Good to see it is now obeying the specified order, with the first thing listed activated.